URLTester/Test/RedirectTest.cs

```diff
@@ -72,18 +72,24 @@ public bool LoadFile()
         /// Execute the list of url test that have been setup
         /// </summary>
         /// <returns>True if no errors occurred / False if an error has occurred during testing.</returns>
-        public virtual bool TestLinks()
+        public virtual bool TestLinks(OutputProgressHandler  handler)
         {
             ErrorMessages = new List<ErrorMessage>();
             var returnValue = true;
             
+            var i = 0;
+            var totalCount = UrlList.Count;
             foreach (var item in UrlList)
             {
+                handler?.Invoke(i, totalCount, item.Url);
+
```

Comment:
At this point a better way to handle this would be to remove your new delegate, introduce an "Events" class and create a new "LinkTested" event that you fire from this location everytime you finish testing a link.  Once the event fires all subscribers of the event would trigger their logic. 

```suggestion
  public class Events
    {
        public delegate void LinkTestedHandler(object sender, EventArgs e);

        public class LinkTestedEventArgs : EventArgs
        {
            public UrlData Item { get; set; }
            public bool Result { get; set; }
        }

    }

testManager.RedirectTest.LinkTested += UpdateProgressBar;
testManager.RedirectTest.LinkTested += OutputStuff;
testManager.RedirectTest.LinkTested += OtherStuffDependentOnThisEvent;
```